### DISCUSSION TITLE
Any way to NOT upload a JPG if an HEIC already exsists with same file name ?

### DISCUSSION THREAD

**Discussion by vw-kombi** [simulot/immich-go#437] [General]
This stems from an old workflow where I would use software to pull the photos of our iphone's and I had it set to convert HEIC to JPG as it imported.  The photos are still HEIC on iphone, and they then have a jpg version in the filesystem on the PC.

I would like immich-go to somehow ignore the file extensions on import if there is one in immich of that name there already ?

i.e :

1 - run immich-go over my icloudPD photo's (these are copied from the iphone in HEIC formast) - meaning all HEIC's are uploaded to immich.
2 - run immich-go over all my file system photos, with some sort of flag to ONLY check the file name, for example IMG5428.JPG is not uploaded because IMG5428.HEIC is already in immich.

This would fix my 30,000+ duplicates in immich - I would delete all of immich and start it all again if immich-go could do the above.  

I tried the de-duplicate thing in immich but too many there to manually do and the automatic cant be done as it PRIORITISES JPG over HEIC as it thinks they are better as they are bigger but that is not the case.  Ideally a setting in immich to prioritise HEIC over JPG which it does not have (I have requested it).

**Comment by simulot** [comment]
Not at the moment.
I have tried to stack them in immich, but this is not satisfying.

That could be a good feature either when importing images or for removing duplicates when already duplicated. 
Stay tuned.

**Comment by vw-kombi** [comment]
I have reported a similar need for my workflow in immich and immich-go in the past.
My previous workflow has meant I have heic's in icloud/on phone, and a jpg version in my filesystem from when i used to copy from the iphone and convert from heic.  Plus I copied random sets of jpg's from the file system to icloud so i could have them on my phone also (once I purchased icloud due to the space saving).

Now I want pure icloud (for ones kept on iphone) and immich backend for these and everything else.
20000+ dups when I added filesystem, then google photos, then icloud.
I am waiting for a feature to either prioritise the heic duplicates in immich itself when mass deleting, or immich-go to manage this in the import process (i.e prioritise heic to import over a matching jpg, and/or only imprt heic if a same jpg exists.
Once I get one of these, I will re reinstalling immich and starting it all again.

**Reply by simulot** [reply]
Thanks for this suggestion! 
I think that both options can be implemented:
- At import time, upload the HEIC (or the RAW) version, and discard the JPG
- With an "cleaning" command that detect the pairs of files and keep the better version

Stay tuned
